### PR TITLE
feat: add TypeScript support

### PR DESCRIPTION
**Pull request by satya164**

### CODE REVIEW

File: test/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "license": "MIT",
   "dependencies": {
-    "@callstack/eslint-config": "*",
+    "@callstack/eslint-config": "link:..",
```
Comment (thymikee) [inline test/package.json]: Neat!

**Comment by thymikee** [comment]
Looks like tests didn't run, could you investigate?

**Comment by satya164** [comment]
@thymikee maybe there's some setting needed? I can even find https://travis-ci.org/callstack/eslint-config-callstack

**Comment by thymikee** [comment]
Need to check that, maybe somebody played with permissions

**Comment by thymikee** [comment]
Forgot I disabled builds for PRs. Fixed.

**Comment by thymikee** [comment]
Just published 4.1.0, thanks @satya164!

**Pull request merged**